### DISCUSSION TITLE
layout for api

### DISCUSSION THREAD

**Discussion by Carlos-err406** [sveltejs/kit#11873] [Q&A] [answered]
hi
is there a way to do some logic for a grouped set of +server routes?
for example say there is a 

/routes/api/(protected)/users/+server.ts
/routes/api/(public)/login/+server.ts

and all routes within the (protected) group have a authentication step before actually accessing the endpoint
how i imagined it was placing a +layout.server in the (protected) group
that takes the request and authenticates it before getting to the actual endpoint
even better would be if there was a way to send data from the layout to the children +server(s)
similar to the PageData or LayoutData

**Comment by kyllerss** [comment, marked as answer]
Look into [server hooks](https://kit.svelte.dev/docs/hooks#server-hooks). You can have the hook redirect the user to the authentication page if they are not authenticated, or let the request go through if they are.
